File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    """
    starting from the top right corner if > target  move left else move below
    
    """
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        N,M = len(matrix),len(matrix[0])
        i,j = 0,M-1
        while i<N and j>-1:
            if matrix[i][j] == target:
                return True
            elif matrix[i][j] < target:
                i += 1
            else:
                j -= 1
        return False

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
/*
    starting from the top right corner if > target  move left else move below
*/
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int N=matrix.size(),M=matrix[0].size();
        int i=0,j=M-1;
        while (j>-1 & i<N){
            if ( matrix[i][j] == target ) return true;
            else if (matrix[i][j]>target) j--;
            else i++;
        }
        return false;
    }
};